net-libs/libs3/libs3-2.0.recipe

```diff
@@ -1,22 +1,72 @@
-DESCRIPTION="C Library API for Amazon S3"
-HOMEPAGE="http://libs3.ischo.com.s3.amazonaws.com/index.html"
-SOURCE_URI="http://libs3.ischo.com.s3.amazonaws.com/libs3-2.0.tar.gz"
-CHECKSUM_MD5="e52da69ddc11019e98cf8246fc55b4e1"
+SUMMARY="C Library API for Amazon S3"
+DESCRIPTION="The libs3 project implements a complete C library API for \
+Amazon S3. It is intended to be easy to use, comprehensive, and suitable \
+for use from any C or C++ program."
+HOMEPAGE="https://github.com/bji/libs3"
+COPYRIGHT="2008 Bryan Ischo"
+LICENSE="GNU GPL v3"
 REVISION="1"
-STATUS_HAIKU="stable"
-DEPEND="dev-libs/libxml2>=2.7.3
-	net-misc/curl>=7.18.2"
+SOURCE_URI="http://archive.ubuntu.com/ubuntu/pool/main/libs/libs3/libs3_2.0.orig.tar.gz"
+CHECKSUM_SHA256="c99857bfbcc9ff47742119bc32f391f38a24acf40cf49902ef3b18b1f0e8c16d"
+PATCHES="libs3-$portVersion.patch"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+libVersion="2"
+libVersionCompat="$libVersion compat >= 2"
+
+PROVIDES="
+	libs3$secondaryArchSuffix = $portVersion
+	cmd:s3$secondaryArchSuffix
+	lib:libs3$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libs3${secondaryArchSuffix}_devel = $portVersion
+	devel:libs3$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	libs3$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcurl$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage libs3$secondaryArchSuffix \
+	"$binDir"/s3 \
+	"$libDir"/libs3.so.$libVersion
 
 BUILD()
 {
-	cd libs3-2.0
-	make DESTDIR=${DESTDIR}/`finddir B_COMMON_DIRECTORY`
+	make $jobArgs
 }
 
 INSTALL()
 {
-	cd libs3-2.0
-	make install DESTDIR=${DESTDIR}/`finddir B_COMMON_DIRECTORY`
+	install -d -m 755 $binDir $libDir $includeDir
+	install -m 755 -t $binDir ./build/bin/s3
+	install -m 755 -t $libDir ./build/lib/libs3.*
```

Comment:
Please note that upstream [install][0] the shared library as `libs3.so.2.0`, so you might want to replicate that. Also the static library should be installed without executable permission.

```diff
-	install -m 755 -t $libDir ./build/lib/libs3.*
+	install -m 755 ./build/lib/libs3.so.${libVersion%.*} $libDir/libs3.so.$libVersion
+	ln -s $libDir/libs3.so.$libVersion $libDir/libs3.so.${libVersion%.*}
+	install -m 644 -t $libDir ./build/lib/libs3.a
```

[0]: https://github.com/bji/libs3/blob/2.0/GNUmakefile#L164